src/Microsoft.ApplicationInsights.AspNet/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -104,6 +110,12 @@ public static IConfigurationBuilder AddApplicationInsightsSettings(this IConfigu
                 wasAnythingSet = true;
             }
 
+            if (level != null)
+            {
+                TelemetryLogger.Instance.Level = level.Value;
+                wasAnythingSet = true;
```

Comment:
we are not adding anything to telemetryConfigurationSource. Please let me know if we still need to set wasAnythingSet to be true?